### PR TITLE
Reject SPIR-V that applies void to OpUndef, OpCopyObject, OpPhi

### LINKED ISSUE #4035: spirv-val: Disallow OpUndef, OpCopyObject and OpPhi on void type
It does not seem sensible to allow void values in these contexts.

### PR DESCRIPTION
**Pull request by afd**
Fixes #4035.

### CODE REVIEW

File: source/opt/inline_pass.cpp [hunk 1]
```diff
@@ -620,13 +620,19 @@ bool InlinePass::GenInlineCode(
...
             call_inst_itr->dbg_line_inst(), call_inst_itr->GetDebugScope());
   } else {
-    // Even though it is very unlikely, it is possible that the result id of
-    // the void-function call is used, so we need to generate an instruction
-    // with that result id.
-    std::unique_ptr<Instruction> undef_inst(
-        new Instruction(context(), SpvOpUndef, call_inst_itr->type_id(),
-                        call_inst_itr->result_id(), {}));
-    context()->AddGlobalValue(std::move(undef_inst));
+    // If the function does not return a value, there could be OpName
```
Comment (alan-baker) [inline source/opt/inline_pass.cpp:623]: This misses decorations, but also I think this is unnecessary. When the function call is removed it should remove all associated names and decorations.

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: source/opt/inline_pass.cpp (hunk 1 above)
Comment (afd) [reply source/opt/inline_pass.cpp:623]: Hmm - it doesn't.  I changed `InlineCallWithStructArgContainingSampledImage` in inline_opaque_test to give a name to the result id of a call, and without the code I added here the name doesn't get removed and the module becomes invalid.

It wasn't that clear to me where in the inliner the function call instruction actually gets killed.  `GenInlineCode` has a `call_inst_itr` parameter which is the call instruction, but I don't see where this instruction gets explicitly eliminated.

Comment (alan-baker) [reply source/opt/inline_pass.cpp:623]: Exhaustive inlining seems to call KillInst. We should probably unify how the various inliners deal with this.

Comment (afd) [reply source/opt/inline_pass.cpp]: OK, I have unified this by moving the code that kills names and decorations into GenInlineCode, which is called by both inliners.  That indeed avoids the need for the extra code above.

**Review by alan-baker — APPROVED** [review]

**Pull request merged**